internal/controller/controllers/agent_controller.go

```diff
@@ -204,6 +210,132 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 	return r.updateStatus(ctx, log, agent, &h.Host, h.ClusterID, nil, false)
 }
 
+func findMatchingNode(agent *aiv1beta1.Agent, nodes []corev1.Node) *corev1.Node {
+	addresses := make([]string, 0)
+	for _, intf := range agent.Status.Inventory.Interfaces {
+		for _, ip := range append(intf.IPV4Addresses, intf.IPV6Addresses...) {
+			splitIp := strings.Split(ip, "/")
+			if len(splitIp) == 2 {
+				addresses = append(addresses, splitIp[0])
+			}
+		}
+	}
+	for i := range nodes {
+		n := &nodes[i]
+		if strings.EqualFold(n.Name, agent.Spec.Hostname) {
```

Comment:
Some comment as below w.r.t the use of `agent.Spec.Hostname`

Reply:
OK